Assets/Scripts/Combat/Abilities/WheelHandler.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class WheelHandler : MonoBehaviour, IPointerDownHandler, IBeginDragHandler, IDragHandler, IEndDragHandler
+{
+    private RectTransform rectTransform;
+    private Quaternion startRotation;
+    private Quaternion lastRotation;
+    float lastAngle;
+    bool dragged;
+    bool fullRotation;
+    int rotations;
+    
+    private void Start() 
+    {
+        rectTransform = GetComponent<RectTransform>();
```

Comment:
double check for each of your Start() methods, that the parameters that you initialize are RE-initialized before every turn. Start() is called once at the beginning of the scene

Reply:
A lot of the code for WheelHandler isn't done since I changed the second phase to the Slider. Right now all it does is rotate (doesn't even check if a full rotation is done, that code doesnt work...)